app/controllers/application_controller.rb

```diff
@@ -10,20 +10,20 @@ class ApplicationController < ActionController::Base
   # def home; end
 
   def trips
-    @trips = policy_scope(Trip).order(created_at: :desc)
+    @trips = policy_scope(Trip).order(:start_date)
     if current_user
-      @my_trips = current_user.trips
+      @my_trips = []
       @trips.each do |trip|
         trip.invitations.each do |invitation|
-          @my_trips << invitation.trip if invitation.user == current_user
+          @my_trips << invitation.trip if trip.user == current_user ||invitation.user == current_user
```

Comment:
I think this here is the issue, maybe try find_or_create_by 
https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by
or something like that

Reply:
Sorry for the oversight. It's missing `unless @my_trips.include? trip`. Added in the new commit.